distribution/addon_distribution.py

```diff
@@ -0,0 +1,244 @@
+import os
+from enum import Enum
+from abc import abstractmethod
+import attr
+import logging
+import requests
+import platform
+
+from distribution.file_handler import RemoteFileHandler
+
+
+class UrlType(Enum):
+    HTTP = "http"
+    GIT = "git"
+    FILESYSTEM = "filesystem"
+
+
+class UpdateState(Enum):
+    EXISTS = "exists"
+    UPDATED = "updated"
+    FAILED = "failed"
+
+
+@attr.s
+class MultiPlatformPath(object):
+    windows = attr.ib(default=None)
+    linux = attr.ib(default=None)
+    darwin = attr.ib(default=None)
```

Comment:
I am wondering wheter this should use `pathlib.Path` type directly

Reply:
What would be benefit? Does it handle something like `source["path"].get(platform.system().lower())` ?